docs/reference/query-languages/esql/esql-commands.md

```diff
@@ -724,10 +728,20 @@ FROM app_logs
 | LOOKUP JOIN service_owners ON service_id
 ```
 
-In case of name collisions, the newly created columns will override existing columns.
+`LOOKUP JOIN` is generally faster when there are fewer rows to join with. {{esql}} will try and perform any `WHERE` clause before the `LOOKUP JOIN` where possible.
+
+The two following examples will have the same results. The two examples have the `WHERE` clause before and after the `LOOKUP JOIN`. It does not matter how you write your query, our optimizer will move the filter before the lookup when ran.
```

Comment:
nit: The optimization is not always possible (e.g. when the filter is on looked up columns, you're out of luck).
```suggestion
The two following examples will have the same results. The two examples have the `WHERE` clause before and after the `LOOKUP JOIN`. It does not matter how you write your query, our optimizer will move the filter before the lookup when possible.
```